posydon/binary_evol/singlestar.py

```diff
@@ -137,8 +137,10 @@ def __init__(self, **kwargs):
         # these quantities are updated in step_SN.py
         if not hasattr(self, 'natal_kick_array'):
             self.natal_kick_array = [None] * 4
-        if not hasattr(self, 'spin_orbit_tilt'):
-            self.spin_orbit_tilt = None
+        if not hasattr(self, 'spin_orbit_tilt_first_SN'):
```

Comment:
Is this nomenclature really needed? We do not allow to have references of another star in a single stellar object. The way of keeping track of this is in the step SN, where you access the S1 and S2 thorugh the binary object.

Reply:
oh, maybe I understood wrong. You want to store the initial tilt and then the updated one after the secod tilt. But then how would the user know witch one to use, e.g., to compute the chi_eff.

Reply:
To be more explicit on the naming, we can may include an after there, hence `spin_orbit_tilt_after_first_SN` and `spin_orbit_tilt_after_second_SN`.
Btw. I just thought of a possible science case for using the the tilt after the first SN: How about mergers of NS/BH with a WD/sdb star? Here we should get only one SN, hence only one tilt. Those could be potential LISA sources beside the WD+WD mergers. Similarly, I can imaging to include AIC in the future, where we have two NS merging, but only one went through a SN.
An other option to make it easy for a user could be to introduce a variable called `final_spin_orbit_tilt`, which gets updated during the evolution.

Reply:
I've updated this as @mkruckow suggested with spin_orbit_tilt_after_first_SN etc.